src/posix.mak

```diff
@@ -494,8 +497,11 @@ install: all
 
 ######################################################
 
-checkwhitespace: $(HOST_DMD_PATH)
-	CC=$(HOST_CXX) $(HOST_DMD_RUN) -run checkwhitespace.d $(SRC) $(GLUE_SRC) $(ROOT_SRCS)
+checkwhitespace: $(HOST_DMD_PATH) $(TOOLS_DIR)
+	CC=$(HOST_CXX) $(HOST_DMD_RUN) -run $(TOOLS_DIR)/checkwhitespace.d $(SRC) $(GLUE_SRC) $(ROOT_SRCS)
+
+$(TOOLS_DIR):
```

Comment:
There are issues with making something depend on a dir. E.g. any (unrelated) change to the dir triggers a rebuild etc.

So I suggest the following:

1. Make the binary depend on `$(TOOLS_DIR)/checkwhitespace.d`
2. Make `$(TOOLS_DIR)/checkwhitespace.d` depend on nothing and fetch it from github.